### PR TITLE
8258018: Remove arrayOop.inline.hpp

### PR DESCRIPTION
**Pull request by iklam**
Please review this trivial change. arrayOop.inline.hpp is not needed and its contents can be safely moved into arrayOop.hpp.

Testing with mach5 tier1,builds-tier2,builds-tier3,builds-tier4,builds-tier5

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8258018](https://bugs.openjdk.java.net/browse/JDK-8258018): Remove arrayOop.inline.hpp

### Reviewers
 * [Stefan Karlsson](https://openjdk.java.net/census#stefank) (@stefank - **Reviewer**) ⚠️ Review applies to 3fff850575c605aea657027390d373ed0701c352
 * [Kim Barrett](https://openjdk.java.net/census#kbarrett) (@kimbarrett - **Reviewer**) ⚠️ Review applies to 3fff850575c605aea657027390d373ed0701c352
 * [Coleen Phillimore](https://openjdk.java.net/census#coleenp) (@coleenp - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1728/head:pull/1728`
`$ git checkout pull/1728`

### CODE REVIEW

**Review by stefank — APPROVED** [review]
Thanks for cleaning this up!

File: src/hotspot/share/oops/arrayOop.hpp [hunk 1]
```diff
@@ -90,7 +90,9 @@ class arrayOopDesc : public oopDesc {
 
   // Returns the address of the first element. The elements in the array will not
   // relocate from this address until a subsequent thread transition.
-  inline void* base(BasicType type) const;
+  inline void* base(BasicType type) const {
```
Comment (kimbarrett) [inline src/hotspot/share/oops/arrayOop.hpp:93]: Drop the unnecessary "inline".

**Review by kimbarrett — APPROVED** [review]

**Review by coleenp — APPROVED** [review]
LGTM.

**Comment by iklam** [comment]
Thanks @stefank @coleenp @kimbarrett for the review. Mach5 tests passed for tier1,builds-tier2,builds-tier3,builds-tier4,builds-tier5
/integrate

**Pull request closed without merging**